impl/src/main/java/org/ehcache/impl/internal/store/heap/OnHeapStore.java

```diff
@@ -720,6 +728,10 @@ public K getKey() {
           }
         });
         cachedValue = backEnd.putIfAbsent(key, fault);
+
+        // Hook added for testing
+        beforeCompletingTheFault();
```

Comment:
I tried but the `synchronized` in the fault is the only thing that happens right after. The only other solution would be to decorate the backEnd. But it means getting it by reflection, wrapping it and setting it back. This seemed the easiest and cleanest.